R/process_sc_all_care_home.R

```diff
@@ -29,54 +29,44 @@ process_sc_all_care_home <- function(
     write_to_disk = TRUE) {
   ## Data Cleaning-----------------------------------------------------
   ch_clean <- data %>%
-    dplyr::mutate(ch_admission_date = fix_sc_start_dates(
-      .data$ch_admission_date,
-      .data$period_start_date
-    )) %>%
-    dplyr::group_by(
-      social_care_id,
-      sending_location,
-      ch_admission_date
-    ) %>%
-    dplyr::mutate(episode_max_discharge_date = max(
-      pmin(period_end_date,
-        ch_discharge_date,
-        na.rm = TRUE
+    dplyr::mutate(
+      record_date = end_fy_quarter(.data[["period"]]),
+      qtr_start = start_fy_quarter(.data[["period"]]),
+      # Set missing admission date to start of the submitted quarter
+      ch_admission_date = dplyr::if_else(
+        is.na(.data[["ch_admission_date"]]),
+        .data[["qtr_start"]],
+        .data[["ch_admission_date"]]
+      ),
+      # TODO check if we should set the dis date to the end of the period?
+      # If the dis date is before admission, remove the dis date
+      ch_discharge_date = dplyr::if_else(
+        .data[["ch_admission_date"]] > .data[["ch_discharge_date"]],
+        lubridate::NA_Date_,
+        .data[["ch_discharge_date"]]
       )
-    )) %>%
-    dplyr::ungroup() %>%
-    dplyr::mutate(test = ifelse(ch_admission_date > ch_discharge_date, 1, 0)) %>%
-    # dplyr::mutate(ch_discharge_date = fix_sc_missing_end_dates(
-    #   .data$ch_discharge_date,
-    #  .data$period_end_date
-    #  )) %>%
-    # Fix ch_discharge_date is earlier than ch_admission_date by setting end_date to the end of fy
-    dplyr::mutate(ch_discharge_date = fix_sc_end_dates(
-      .data$ch_admission_date,
-      .data$ch_discharge_date,
-      .data$period
-    )) %>%
+    ) %>%
     dplyr::left_join(sc_demog_lookup,
       by = c("sending_location", "social_care_id")
     ) %>%
     replace_sc_id_with_latest()
```

Comment:
Thank you for updating the code! I'm not sure if my point is relevant so feel free to leave it!

I noticed in the resolved PR "Deal with NA in ch_provider and missing group_by()", this function is used later in the script (found on lines 89 and 90) and passes fixed_ch_provider through this function to create a new object fixed_sc_id. fixed_sc_id then feeds into fixed_nursing_provision

Reply:
Fixed this - we have reverted back to master copy